app/controllers/sessions.rb

```diff
@@ -0,0 +1,19 @@
+get '/session/new' do
+  @user = User.new
+  erb :'/session/new'
+end
+
+post '/session/new' do
+  @info = params[:user]
+  if session_authenticate(@info[:name], @info[:password])
+    redirect("/")
+  else
+    redirect('/session/new')
```

Comment:
You should render `erb :'/session/new'` right away when authentication fails. Use the @user object to display errors.